### PR TITLE
Remove unused malloc.h include which throws warning on Linux/macOS.

### PR DESCRIPTION
**Pull request by nmoinvaz**

### CODE REVIEW

File: steamapi.cpp [hunk 1]
```diff
@@ -1,6 +1,5 @@
-#include <string.h>
-#include <malloc.h>
 #include <stdlib.h>
+#include <string.h>
```
Comment (sergio-nsk) [inline steamapi.cpp:1]: Should be 
```cpp
#include <cstdlib>
#include <cstring>
```

File: steamapps.cpp [hunk 2]
```diff
@@ -1,6 +1,5 @@
-#include <string.h>
-#include <malloc.h>
 #include <stdlib.h>
+#include <string.h>
```
Comment (sergio-nsk) [inline steamapps.cpp:1]: Should be 
```cpp
#include <cstdlib>
#include <cstring>
```

File: steamfriends.cpp [hunk 3]
```diff
@@ -1,6 +1,5 @@
-#include <string.h>
-#include <malloc.h>
 #include <stdlib.h>
+#include <string.h>
```
Comment (sergio-nsk) [inline steamfriends.cpp:1]: Should be 
```cpp
#include <cstdlib>
#include <cstring>
```

File: steamuser.cpp [hunk 4]
```diff
@@ -1,6 +1,5 @@
-#include <string.h>
-#include <malloc.h>
 #include <stdlib.h>
+#include <string.h>
```
Comment (sergio-nsk) [inline steamuser.cpp:1]: Should be 
```cpp
#include <cstdlib>
#include <cstring>
```

File: steamuserstats.cpp [hunk 5]
```diff
@@ -1,6 +1,5 @@
-#include <string.h>
-#include <malloc.h>
 #include <stdlib.h>
+#include <string.h>
```
Comment (sergio-nsk) [inline steamuserstats.cpp:1]: Should be 
```cpp
#include <cstdlib>
#include <cstring>
```

File: steamutils.cpp [hunk 6]
```diff
@@ -1,6 +1,5 @@
-#include <string.h>
-#include <malloc.h>
 #include <stdlib.h>
+#include <string.h>
```
Comment (sergio-nsk) [inline steamutils.cpp:1]: Should be 
```cpp
#include <cstdlib>
#include <cstring>
```

**Review by sergio-nsk — CHANGES_REQUESTED** [review]

File: steamapi.cpp (hunk 1 above)
Comment (nmoinvaz) [reply steamapi.cpp:1]: I don't have any intention of updating the other headers. I just reordered them. I will fix it so only malloc.h is removed.

**Comment by nmoinvaz** [comment]
I have only removed malloc.h to get it to build on Linux. I don't care about the other headers. So I corrected the PR to remove my reordering of the other includes.

File: steamapi.cpp (hunk 1 above)
Comment (sergio-nsk) [reply steamapi.cpp:1]: You will not only remove `malloc.h`, you will add `string.h`.

File: steamapi.cpp [hunk 7]
```diff
@@ -1,5 +1,4 @@
 #include <string.h>
```
Comment (sergio-nsk) [inline steamapi.cpp:1]: I don't see `#include <string.h>` is necessary in this file. Neither `malloc` nor `str*` CRT functions are used there.

File: steamapi.cpp (hunk 1 above)
Comment (nmoinvaz) [reply steamapi.cpp:1]: I have changed the PR.

File: steamfriends.cpp [hunk 8]
```diff
@@ -1,5 +1,4 @@
 #include <string.h>
```
Comment (sergio-nsk) [inline steamfriends.cpp:1]: I don't see `#include <string.h>` is necessary in this file. Neither `malloc` nor `str*` CRT functions are used there.

File: steamapps.cpp [hunk 9]
```diff
@@ -1,5 +1,4 @@
 #include <string.h>
```
Comment (sergio-nsk) [inline steamapps.cpp:1]: I don't see `#include <string.h>` is necessary in this file. Neither `malloc` nor `str*` CRT functions are used there.

**Review by sergio-nsk — CHANGES_REQUESTED** [review]

File: steamapi.cpp (hunk 7 above)
Comment (nmoinvaz) [reply steamapi.cpp:1]: I'm not sure. It is not really what I wanted to be looking into for this PR.

**Review by sergio-nsk — APPROVED** [review]

File: steamapi.cpp (hunk 7 above)
Comment (nmoinvaz) [reply steamapi.cpp:1]: I will try to look into it when I am done with these other PRs.

**Pull request merged**